FinnUI/Sources/Components/ConfettiView/ConfettiView.swift

```diff
@@ -0,0 +1,103 @@
+//
+//  Copyright © 2020 FINN.no AS. All rights reserved.
+//
+
+import FinniversKit
+
+public class ConfettiView: UIView {
+
+    // MARK: - Private properties
+
+    private let confettiImages: [UIImage] = [
+        .init(named: .confetti1),
+        .init(named: .confetti2)
+    ]
+
+    private let confettiColors: [UIColor] = [
+        .primaryBlue,
+        .secondaryBlue,
+        .pea,
+        .watermelon
+    ]
+
+    // MARK: - Init
+
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        setup()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private func setup() {
+        backgroundColor = .clear
+        isUserInteractionEnabled = false
+    }
+
+    // MARK: - Public methods
+
+    /// Start the creation of confetti
+    ///
+    /// Call this method to start the creation of confetti
+    /// - Warning: `completion` is called when the confetti creation is completed, not when the last confetti is off screen
+    /// - Parameter duration: how long confetti will be added/created, in seconds
+    /// - Parameter completion: called when the confetti creation is completed.
+    public func start(withDuration duration: TimeInterval = 0.75, completion: (() -> Void)? = nil) {
+        let emitterLayer = createConfettiEmitterLayer()
+
+        layer.addSublayer(emitterLayer)
+
+        DispatchQueue.main.asyncAfter(
+            deadline: .now() + duration,
+            execute: {
+                emitterLayer.lifetime = 0
+                completion?()
+            }
+        )
+    }
+
+    // MARK: - Private methods
+
+    private func createConfettiEmitterLayer() -> CAEmitterLayer {
+        let emitterLayer = CAEmitterLayer()
+        emitterLayer.emitterSize = CGSize(width: bounds.size.width, height: 1)
+        emitterLayer.emitterPosition = CGPoint(x: bounds.size.width / 2, y: -50)
+        emitterLayer.emitterShape = .line
+
+        let beginTime = CACurrentMediaTime()
+
+        emitterLayer.emitterCells = (0...8).map {
+            let color = confettiColors[$0 % confettiColors.count]
+            return createConfettiEmitterCell(beginTime: beginTime, color: color)
+        }
+
+        return emitterLayer
+    }
+
+    private func createConfettiEmitterCell(beginTime: CFTimeInterval, color: UIColor) -> CAEmitterCell {
+        let cell = CAEmitterCell()
+
+        cell.beginTime = beginTime
+        cell.birthRate = 10
+        cell.lifetime = 5
+
+        cell.contents = confettiImages.randomElement()?.cgImage
+        cell.color = color.cgColor
+
+        cell.emissionLongitude = CGFloat.pi
+        cell.emissionRange = CGFloat.pi / 8
+
+        cell.velocity = UIDevice.current.userInterfaceIdiom == .pad ? 600 : 500
```

Comment:
You could use `UIDevice.isIPad()`. We try not to have the isIPad-check, but in this case it doesn't make sense to use `traitCollection.horizontalSizeClass` instead, since the vertical size is what matters here, I guess? 😄

Reply:
there is a verticalSizeClass too :)

Reply:
Right, that could work, I haven't tested it before though 🤔

Reply:
Yes, I thought about it. It made sense to me to differentiate for both size classes (even though we don't support the landscape orientation), so that's why I just checked for `pad`.

Reply:
But in retrospect, maybe I should just have one velocity 🤔

Reply:
The 2 hardest things in programming:
1) Naming
2) Deciding which velocities to use for your confetti-rain

:trollface: